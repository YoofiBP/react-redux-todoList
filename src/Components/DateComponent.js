import React from 'react';
import IconButton from "@material-ui/core/IconButton";
import ArrowForwardIosIcon from '@material-ui/icons/ArrowForwardIos';
import ArrowBackIosIcon from '@material-ui/icons/ArrowBackIos';
import dayjs from 'dayjs'
import { changeDate } from '../redux/actions';
import { connect } from 'react-redux';
import { DATE_BACK, DATE_FORWARD } from '../redux/actionTypes';

function DateComponent(props){
    let today = (props.currentDate === dayjs().format("MM/DD/YYYY")) ? "Today" : dayjs(props.currentDate).format('ddd, MMM D, YYYY')
    return (
        <>
        <IconButton
        aria-label="delete"
        onClick={() => props.changeDate(DATE_BACK)}
        edge="end"
        className="p-2"
        size="small"
        id="back"
      >
        <ArrowBackIosIcon />
      </IconButton>
      <h1>{today}</h1>
      <IconButton
        aria-label="delete"
        onClick={() => props.changeDate(DATE_FORWARD)}
        edge="end"
        className="p-2"
        size='small'
        id="forward"
      >
        <ArrowForwardIosIcon />
      </IconButton>
      </>
    )
}

const mapStateToProps = (state) => {
  return {currentDate: state.date }
}

const mapDispatchToProps = dispatch => {
  return {
    changeDate: direction => {
      dispatch(changeDate(direction));
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(DateComponent);