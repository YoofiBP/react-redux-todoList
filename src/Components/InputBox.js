import React from 'react';
import { connect } from 'react-redux';
import { addItem } from '../redux/actions';
//import { users } from '../users';

class InputBox extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            value:""
        }
        this.handleChange = this.handleChange.bind(this)
        this.handleSubmit = this.handleSubmit.bind(this)
    }

    handleChange(event){
        this.setState({value: event.target.value})
    }

    handleSubmit(event){
        if(this.state.value.length > 1){
        this.props.addNewTodo(this.state.value, this.props.currentDate)
        this.setState({value:""})
        }
        event.preventDefault();
    }

    render(){
    return (
        <div className="input-group mb-2">
            <form className="input-group mb-2 w-100" onSubmit={this.handleSubmit}>
  <input id="title" onChange={this.handleChange} type="text" className="form-control" value={this.state.value} placeholder="New Item" aria-label="New Item" aria-describedby="button-addon2"/>
  <input type="submit" className="btn btn-outline-secondary" value="Add Item" />
  </form>
</div>
    )
    }
}

export class Welcome extends React.Component{
    constructor(props){
        super(props);
        this.state = {user: {}}
    }

    componentDidMount() {
    fetch('/users.json')
    .then(res => res.json())
    .then(users => { return users.filter(user => user.userId === this.props.userId)})
    .then(user => this.setState({user: user[0]}))
    .catch(err => console.log(err))
    }

    render() {
        return (
            <h1>Welcome {this.state.user.username}!</h1>
        )
    }
    
}

const mapDispatchToProps = dispatch => {
    return {
        addNewTodo: (title, date) => {
            dispatch(addItem(title, date))
        }
    }
}

const mapStateToProps = state => {
    return {
        currentDate: state.date
    }
}   

export default connect(mapStateToProps, mapDispatchToProps)(InputBox);