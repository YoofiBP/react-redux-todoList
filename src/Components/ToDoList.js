import React from 'react';
import TodoListItem from './ToDoListItem'
import { connect } from 'react-redux'

function ToDoList(props) {
    return (
        props.items.map(item => <TodoListItem item={item} key={item.id} id={item.id}/>)
    )
}

const mapStateToProps = state => {
    let [todaysItems] = state.todoList.filter(day => day.date === state.date)
    return todaysItems ? { items: todaysItems.tasks } : { items: [] }
}

export default connect(mapStateToProps)(ToDoList);