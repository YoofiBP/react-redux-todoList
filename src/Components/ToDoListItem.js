import React from "react";
import IconButton from "@material-ui/core/IconButton";
import DeleteIcon from "@material-ui/icons/Delete";
import InputIcon from "@material-ui/icons/Input";
import dayjs from "dayjs";
import Tooltip from "@material-ui/core/Tooltip";

import { connect } from 'react-redux';
import { addItem, checkItem, deleteItem } from '../redux/actions';

function TodoListItem(props) {
  const { title, completed } = props.item;
  return (
    <div className="d-flex">
      <li>
        <input
          type="checkbox"
          value={title}
          checked={completed}
          className="mr-auto p-2"
          onChange={() => props.checkItem(props.id,props.currentDate)}
        />{" "}
        <label
          className="mr-auto p-2"
          style={
            completed
              ? { textDecoration: "line-through", color: "grey" }
              : { textDecoration: "none" }
          }
        >
          {title}
        </label>
        <Tooltip title="Delete">
          <IconButton
            aria-label="delete"
            onClick={() => props.deleteItem(props.id, props.currentDate)}
            edge="end"
            className="p-2"
            id="delete"
          >
            <DeleteIcon />
          </IconButton>
        </Tooltip>
        <Tooltip title="Do Tomorrow">
          <IconButton onClick={() => props.moveItem(props.id, props.currentDate, title)} id="move">
            <InputIcon />
          </IconButton>
        </Tooltip>
      </li>
    </div>
  );
}

const mapDispatchToProps = dispatch => {
  return {
    checkItem: (index, date) => {
      dispatch(checkItem(index, date))
    },
    deleteItem: (index, date) => {
      dispatch(deleteItem(index, date))
    },
    moveItem: (index, date, title) => {
      dispatch(deleteItem(index, date));
      date = dayjs(date, "MM/DD/YYYY").add(1, "day").format("MM/DD/YYYY")
      dispatch(addItem(title, date))
    }
  }
}

const mapStateToProps = state => {
  return {
      currentDate: state.date
  }
} 
export default connect(mapStateToProps, mapDispatchToProps)(TodoListItem);