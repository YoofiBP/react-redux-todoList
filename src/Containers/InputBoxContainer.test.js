import React from 'react';
import {InputBox} from '../Components/InputBox'
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';


describe('InputBox Component tests', () => {
  it('shallow renders without crashing', () => {
    shallow(<InputBox />)
  })

  it('InputBox component matches snapshot', () => {
    const inputBox = create(<InputBox />);
    const JSON = inputBox.toJSON();
    expect(JSON).toMatchSnapshot();
  })

  it("Renders one form element with text input box and submit button", () => {
    const inputBoxWrapper = shallow(<InputBox />);
    expect(inputBoxWrapper.find('form')).toHaveLength(1);
    const expected = ['text', 'submit'];
    const types = inputBoxWrapper.find('input').map((node) => node.prop('type'))
    expect(types).toEqual(expect.arrayContaining(expected));
  })

  it("Has default state", () => {
    const inputBoxWrapper = shallow(<InputBox />);
    expect(inputBoxWrapper.state('value')).toBeFalsy();
  })

  it("Updates value state", () => {
    const inputBoxWrapper = shallow(<InputBox />);
    const event = {
      target: {
        value: "JollyRancher"
      }
    }
    inputBoxWrapper.instance().handleChange(event)
    expect(inputBoxWrapper.state('value')).toEqual("JollyRancher");
  })

  it("Validates and submits value", () => {
    const onClick = jest.fn();
    const inputBoxWrapper = shallow(<InputBox onClick={onClick}/>);
    let event = {
      target: {
        value: "J"
      },
      preventDefault: () => {}
    }
    inputBoxWrapper.instance().handleChange(event)
    inputBoxWrapper.instance().handleSubmit(event);
    expect(onClick).not.toHaveBeenCalled();

    event = {
      target: {
        value: "Testing with Jest"
      },
      preventDefault: () => {}
    }
    inputBoxWrapper.instance().handleChange(event)
    inputBoxWrapper.instance().handleSubmit(event);
    expect(onClick).toHaveBeenCalled();
    expect(inputBoxWrapper.state('value')).toBeFalsy();
  })

  it('Passes functions to appropriate children', () => {
    const inputBoxWrapper = shallow(<InputBox />);
    const handleChange = inputBoxWrapper.instance().handleChange;
    const handleSubmit = inputBoxWrapper.instance().handleSubmit;
    expect(inputBoxWrapper.find("#title").prop('onChange')).toEqual(handleChange)
    expect(inputBoxWrapper.find('form').prop('onSubmit')).toEqual(handleSubmit)
  })

  it("Binds functions appropriately", () => {
    const onClick = jest.fn();
    const inputBoxWrapper = shallow(<InputBox onClick={onClick}/>);
    const inputfield = inputBoxWrapper.find('#title');
    const formComponent = inputBoxWrapper.find('form');
    const event = {
      target: {
        value: "Testing with Jest"
      },
      preventDefault: () => {}
    }
    inputfield.prop('onChange')(event)
    expect(inputBoxWrapper.state('value')).toEqual('Testing with Jest')
    formComponent.prop('onSubmit')(event)
    expect(onClick).toHaveBeenCalled();
    expect(inputBoxWrapper.state('value')).toBeFalsy();
  })
})

//Items below 4 chars should not call onClick