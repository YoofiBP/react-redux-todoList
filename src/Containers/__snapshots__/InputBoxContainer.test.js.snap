// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox Component tests InputBox component matches snapshot 1`] = `
<div
  className="input-group mb-2"
>
  <form
    className="input-group mb-2 w-100"
    onSubmit={[Function]}
  >
    <input
      aria-describedby="button-addon2"
      aria-label="New Item"
      className="form-control"
      id="title"
      onChange={[Function]}
      placeholder="New Item"
      type="text"
      value=""
    />
    <input
      className="btn btn-outline-secondary"
      type="submit"
      value="Add Item"
    />
  </form>
</div>
`;

exports[`InputBoxContainer tests Matches snapshot 1`] = `
<div
  className="input-group mb-2"
>
  <form
    className="input-group mb-2 w-100"
    onSubmit={[Function]}
  >
    <input
      aria-describedby="button-addon2"
      aria-label="New Item"
      className="form-control"
      id="title"
      onChange={[Function]}
      placeholder="New Item"
      type="text"
      value=""
    />
    <input
      className="btn btn-outline-secondary"
      type="submit"
      value="Add Item"
    />
  </form>
</div>
`;
